openvino_xai/explainer/explanation.py

```diff
@@ -30,14 +31,21 @@ class Explanation:
 
     def __init__(
         self,
-        saliency_map: np.ndarray,
+        saliency_map: np.ndarray | Dict[int | str, np.ndarray],
         targets: np.ndarray | List[int | str] | int | str,
         label_names: List[str] | None = None,
     ):
         targets = convert_targets_to_numpy(targets)
 
-        self._check_saliency_map(saliency_map)
-        self._saliency_map = self._format_sal_map_as_dict(saliency_map)
+        if isinstance(saliency_map, np.ndarray):
+            self._check_saliency_map(saliency_map)
+            self._saliency_map = self._format_sal_map_as_dict(saliency_map)
+            self.total_num_targets = len(self._saliency_map)
+        elif isinstance(saliency_map, dict):
+            self._saliency_map = saliency_map
+            self.total_num_targets = None
```

Comment:
Can you please explain when saliency maps for explanation can be a dict?

Reply:
When it is not required to get saliency maps for all classes, it might be beneficial from efficiency perspective to generate maps only for some classes. For example this is the case with AISE (it generates results per class). For RISE for example, if you check the code, we also construct sparse tensor in the end.

Now, when we generate maps only for some classes, we no longer might pack them as a raw numpy tensor.